trulens_eval/trulens_eval/tru.py

```diff
@@ -720,30 +721,46 @@ def get_records_and_feedback(
 
     def get_leaderboard(
         self,
-        app_ids: Optional[List[mod_types_schema.AppID]] = None
+        app_ids: Optional[List[mod_types_schema.AppID]] = None,
+        group_by_metadata_key: Optional[str] = None
     ) -> pandas.DataFrame:
         """Get a leaderboard for the given apps.
 
         Args:
             app_ids: A list of app ids to filter records by. If empty or not given, all
                 apps will be included in leaderboard.
+            group_by_metadata_key: A key included in record metadata that you want to group results by.
 
         Returns:
             Dataframe of apps with their feedback results aggregated.
+            If group_by_metadata_key is provided, the dataframe will be grouped by the specified key.
         """
 
         if app_ids is None:
             app_ids = []
 
         df, feedback_cols = self.db.get_records_and_feedback(app_ids)
 
-        col_agg_list = feedback_cols + ['latency', 'total_cost']
+        df['meta'] = [
+            json.loads(df["record_json"][i])["meta"] for i in range(len(df))
+        ]
 
-        leaderboard = df.groupby('app_id')[col_agg_list].mean().sort_values(
-            by=feedback_cols, ascending=False
-        )
+        df[str(group_by_metadata_key)] = [
```

Comment:
so if `group_by_metadata_key` is not specified, won't this create a column named `"None"` in the df? 

I am thinking it might be better to but this column creation under the condition only when  `group_by_metadata_key` is present

Reply:
It was creating the column but not using it - but you’re right it would be better to only create it at all if the key is present. Should be good now